### PR TITLE
Make code compatible with TF2 🔥

### PR DESCRIPTION
**Pull request by jeylau**
- Refactored PoseDatasets and PoseNets and made code TensorFlow 2 compatible. 

- **Benchmarked on 4 datasets** (single- and multi-animal, w/ grayscale and color images) with TensorFlow (TF)1.15.5 (which serves as reference), TF2.3, and TF2.5; batch size 8, 30k iterations (except for the marmosets: 20k); 3 backbones (resnet_50, mobilenet_v2_0.5, efficientnet-b0); 2 GPU devices (TITAN RTX & GEFORCE GTX 1080). No significant main effects of either backbone or tf version were found. **Training duration is reported relative to TF1 training time** (Y axis, and value printed above each bar), and in seconds (underneath/within the bar).
profiling

- Not shown here is training with batch size 1 and TF2.5 with a single-animal project (openfield), which is ~4 min (~15%) slower than with TF1. Note that this is not observed with TF2.3.

### CODE REVIEW

**Review by MMathisLab — APPROVED** [review]

**Comment by MMathisLab** [comment]
Not sure why it's failing ... I attempted to pin tables, yet:
```
ERROR: Could not find a version that satisfies the requirement tables==3.6.1 (from versions: 2.3, 2.3.1, 2.4.0, 3.0.0, 3.1.0, 3.1.1, 3.2.0rc1, 3.2.0rc2, 3.2.0, 3.2.1.1, 3.2.2, 3.2.3, 3.2.3.1, 3.3.0, 3.4.0, 3.4.1, 3.4.2, 3.4.3, 3.4.4, 3.5.1, 3.5.2, 3.6.1)
ERROR: No matching distribution found for tables==3.6.1
Error: Process completed with exit code 1.
```

**Comment by jeylau** [comment]
Yeah, that's odd; locally it builds fine with the same settings though (macOS/python3.9.5/updated pip). I also confirm that `conda install -c conda-forge pytables` solves the issue 🤔

**Comment by MMathisLab** [comment]
Alright; order of operations matters here; and seems pytables might not have wheels for the latest 3.9.

That being said, as Jessy mentions above, using the conda file for installation works, as tables is built from conda forge. Tests pass on ubuntu for 3.9. For now, we will pin 3.8 in the conda file but to users, it is compatible with 3.9. 

Screen Shot 2021-06-30 at 1 01 41 PM

Otherwise, all testscripts pass for me on MacOS and Ubuntu 20.04 LTS; for future ubunut 20.04 users, note there can be issues with wxPython wheels, so I made a how-to here: https://github.com/DeepLabCut/Docker4DeepLabCut2.0/wiki/ubuntu-20.04-LTS-fresh-install-guide

Next, I used a TF1 project and analyzed a video without issue in TF2.5; i.e., it's backwards compatible for single animal and latest maDLC projects. Thus, to use TF the only change is a fresh install with the new DEEPLABCUT.yaml is required.

**Review by MMathisLab — APPROVED** [review]

**Review by AlexEMG — APPROVED** [review]
Great job @jeylau!
